<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Computational Techniques in Biological Sequence Analysis | Pablo A. Millan Arias </title> <meta name="author" content="Pablo A. Millan Arias"> <meta name="description" content="A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%8D%81&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://millanp95.github.io/cs482/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Pablo</span> A. Millan Arias </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/courses/">teaching </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Computational Techniques in Biological Sequence Analysis</h1> <p class="post-description"></p> </header> <article> <p><strong>Time:</strong> Mon/Wed 4:00 - 5:20 pm (Except the Feb 15 – 23 week)<br> <strong>Location:</strong> MC 1056<br> <strong>Instructor:</strong> Dr. Pablo Millan Arias <br> Email: pmillana[at]uwaterloo[dot]ca<br> Office: DC 3620 Office hours: Tue 4:00 pm – 5:00 pm or by appointment</p> <p><strong>Teaching assistants:</strong></p> <ul> <li> <p>Michael Wrana<br> Email: mmwrana@uwaterloo.ca<br> Office hours: Mon 9:00 am – 10:00 am (Tentative) DC 3333</p> </li> <li> <p>Jessica Zhu<br> Email: j364zhu@uwaterloo.ca<br> Office hours: Wed 11:00 am – 12:00 pm Station A in MC4065 (CS consulting centre)</p> </li> </ul> <p><strong>Piazza</strong>: <a href="https://piazza.com/uwaterloo.ca/winter2025/cs482682" rel="external nofollow noopener" target="_blank">link</a></p> <p><strong>Course Description:</strong> Due to the large volume of data generated by genome sequencing and cellular measurements of gene expression changes, computer science and mathematics have profoundly changed the science of modern biology. Computational methods are now critical to developing experimental and analytical tools in biological sciences. This course will provide students from all disciplines with an introduction to basic computational skills used to understand and analyze biological data. It combines theories and applications, with a focus on the latter.</p> <p><strong>Main Objectives:</strong> <br> The main aims of the class are for each student to:</p> <ol> <li>Become familiar with different data types used in molecular biology research or biomedicine.</li> <li>Understand the classic algorithms behind the tools so that you can judge which tool is appropriate and which choices should be made during usage.</li> <li>Get a first approximation to modern technologies being used as a substitute for classical algorithms to gain more insight into various biological mechanisms.</li> <li>Become familiar with some of the online tools that are available for analyzing genomic data.</li> <li>Gain practical experience in obtaining data from online resources and analyzing the data within an appropriate biological context.</li> <li>For graduate students, relate the course content to their research.</li> </ol> <p><strong>Format:</strong> The course will be taught in the classroom as lectures. It (tentatively) consists of 12 topics in the area of computational biology, each of which will be covered by one or two classes each week.</p> <p><strong>Prerequisites:</strong> Students are not required to have any prior knowledge of molecular biology, but some understanding of basic biological concepts is beneficial. Because this is an upper-level course, we prefer that students have taken algorithm/data structure-related courses prior to this course and have solid reasonable programming skills. Some basic math and statistics will be covered in the class.</p> <p><strong>Course materials:</strong> Slides (and some videos) presented during the class will be provided to the students through a Learn after the first week of class. Other materials will be made available to the students where applicable. Redistribution of course materials by the student without the instructor’s consent is not permitted.</p> <p><strong>Textbook:</strong> No textbook is required for this class. Computational molecular biology is a rapidly evolving field. Thus, learning materials are widely available and frequently updated on the Internet, in the forms of forum discussions, program documentation and journal papers, in addition to electronic course materials. The students may find the following optional textbooks beneficial if they want to extend their understanding of computational molecular biology, in both breadth and depth, on the topics covered or not covered by the classes.</p> <ul> <li>An Introduction to Applied Bioinformatics (IAB) by J Gregory Caporaso. This is a free, interactive ebook, available at: <a href="http://readiab.org." rel="external nofollow noopener" target="_blank">link</a> </li> <li>Nature Computational Biology compendium which collected a series of tutorials about classic computational biology-related topics, is available at: <a href="https://www.nature.com/collections/tmdlscdqmc" rel="external nofollow noopener" target="_blank">link</a> </li> <li>Bioinformatics Algorithms: An Active Learning Approach (3rd Edition) (2018) by Phillip Compeau and Pavel Pevzner</li> </ul> <p><strong>Grading Scheme:</strong> Programming Assignment: 50% (15% x 4), Midterm Evaluation (20%), Final exam: 40%.</p> <p><strong>Programming Assignments:</strong> The students are expected to complete the assignments on their own. For each assignment, the source code and the assignment report will be reviewed and graded by the TA. The source code should be readable, runnable, and pass the tests. The project report should be detailed enough for the TA to run the code and expected to have the following sections: introduction, implementation details, results, and discussion.<strong>The students are expected to program in Python for programming assignments</strong>.</p> <p><strong>Late Policy:</strong> There is some flexibility in the submission of assignments. Each student is allotted a maximum of 72 extra hours throughout the term to submit the assignments. No justification is required for a late submission. However, no submissions will be accepted after 72 hours. After your total time of 72 hours is consumed, the penalty will be 5% per hour late.</p> <p><strong>Exam:</strong> Students will take a midterm exam and a final exam. The midterm exam will be based on the material covered in the class with knowledge questions as well as programming/design questions. The final exam will be an open book with practical problems and design questions.</p> <p>Students must take the exams independently, without using reference materials or electronic devices. Make-up exams require permission in advance with a doctor’s excuse in the case of illness. COVID-19 policy: Please refer to the UWaterloo Coronavirus website for updated announcements <a href="https://uwaterloo.ca/coronavirus/" rel="external nofollow noopener" target="_blank">https://uwaterloo.ca/coronavirus/</a>. “Starting January 9, a mask requirement will not continue but we strongly encourage you to think of the people around you and help limit the spread of COVID by wearing a mask in indoor settings. As always, we will not hesitate to bring back a requirement to wear masks if the situation requires it at any point during the term.”</p> <h4 id="class-schedule">Class Schedule</h4> <p>This is a <strong>tentative</strong> class schedule and may be adjusted as the term progresses</p> <table> <thead> <tr> <th>Lecture</th> <th>Date</th> <th>Topic</th> </tr> </thead> <tbody> <tr> <td>L01</td> <td>Jan 6</td> <td>Course Overview and Logistics</td> </tr> <tr> <td>L02</td> <td>Jan 8</td> <td>A primer in Computational Biology</td> </tr> <tr> <td>L03</td> <td>Jan 13</td> <td>Sequence Alignment</td> </tr> <tr> <td>L04</td> <td>Jan 15</td> <td>Sequence Database search</td> </tr> <tr> <td>L05</td> <td>Jan 20</td> <td>Sequence Database search / Multiple Sequence Alignment</td> </tr> <tr> <td>L06</td> <td>Jan 22</td> <td>Phylogeny</td> </tr> <tr> <td>*Assignment 1 due Jan 23</td> <td> </td> <td> </td> </tr> <tr> <td>L07</td> <td>Jan 27</td> <td>Genome assembly and mapping</td> </tr> <tr> <td>L08</td> <td>Jan 29</td> <td> </td> </tr> <tr> <td>L09</td> <td>Feb 3</td> <td>Foundations in Deep Learning: Neural Nets, CNN, Autoencoder</td> </tr> <tr> <td>L10</td> <td>Feb 5</td> <td>Alignment free methods Deep Learning in Genomics</td> </tr> <tr> <td>L11</td> <td>Feb 10</td> <td> </td> </tr> <tr> <td> </td> <td>Feb 12</td> <td>Miterm Exam (No Lecture)</td> </tr> <tr> <td>Reading Week</td> <td>Feb 17</td> <td> </td> </tr> <tr> <td> </td> <td>Feb 19</td> <td> </td> </tr> <tr> <td>L12</td> <td>Feb 24</td> <td>Genome Annotation</td> </tr> <tr> <td>L13</td> <td>Feb 26</td> <td> </td> </tr> <tr> <td>*Assignment 2 due Feb 28</td> <td> </td> <td> </td> </tr> <tr> <td>L14</td> <td>Mar 3</td> <td>RNA-Sequencing &amp; Gene Expression</td> </tr> <tr> <td>L15</td> <td>Mar 5</td> <td> </td> </tr> <tr> <td>L16</td> <td>Mar 10</td> <td>RNA &amp; protein structure prediction</td> </tr> <tr> <td>L17</td> <td>Mar 12</td> <td> </td> </tr> <tr> <td>*Assignment 3 due Mar 15</td> <td> </td> <td> </td> </tr> <tr> <td>L18</td> <td>Mar 17</td> <td>Proteomics and Mass Spectrometry</td> </tr> <tr> <td>L19</td> <td>Mar 19</td> <td> </td> </tr> <tr> <td>L20</td> <td>Mar 24</td> <td>Single-cell genomics</td> </tr> <tr> <td>L21</td> <td>Mar 26</td> <td> </td> </tr> <tr> <td>L22</td> <td>Mar 31</td> <td>Foundations in Deep Learning: Transformer Models</td> </tr> <tr> <td>L23</td> <td>Apr 2</td> <td>DNA Language Models</td> </tr> <tr> <td>*Assignment 4 due Apr 4</td> <td> </td> <td> </td> </tr> </tbody> </table> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Pablo A. Millan Arias. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.min.js" integrity="sha256-rjmgmaB99riUNcdlrDtcAiwtLIojSxNyUFdl+Qh+rB4=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> <script src="/assets/js/shortcut-key.js"></script> </body> </html>